hydromt/components/table.py

```diff
@@ -0,0 +1,126 @@
+"""Table component."""
+
+import glob
+import os
+from os.path import basename, dirname, join
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Optional,
+    Union,
+    cast,
+)
+
+import pandas as pd
+
+from hydromt.components.base import ModelComponent
+from hydromt.hydromt_step import hydromt_step
+
+if TYPE_CHECKING:
+    from hydromt.models.model import Model
+
+__all__ = ["TableComponent"]
+
+
+class TableComponent(ModelComponent):
+    """Table Component."""
+
+    def __init__(
+        self,
+        model: "Model",
+    ):
+        """Initialize a TableComponent.
+
+        Parameters
+        ----------
+        model: Model
+            HydroMT model instance
+        """
+        self._data: Optional[Dict[str, Union[pd.DataFrame, pd.Series]]] = None
+        super().__init__(model=model)
+
+    @property
+    def data(self) -> Dict[str, Union[pd.DataFrame, pd.Series]]:
+        """Model tables."""
+        if self._data is None:
+            self._initialize_tables()
+        if self._data is None:
+            raise RuntimeError("Could not load data for table component")
+        else:
+            return self._data
+
+    def _initialize_tables(self, skip_read=False) -> None:
+        """Initialize the model tables."""
+        if self._data is None:
+            self._data = dict()
+            if self._root.is_reading_mode() and not skip_read:
+                self.read()
+
+    @hydromt_step
+    def write(self, fn: str = "tables/{name}.csv", **kwargs) -> None:
```

Comment:
In my PRs I have been trying to make some consts for the file names. I would recommend in the top of this file for a `DEFAULT_FN = "tables/{name}.csv"`